Docs/developer_guide/python_faq.md

```diff
@@ -2,6 +2,85 @@
 
+## What is the Slicer Python Environment?
+
+You can consider each Slicer installation as a virtual Python environment - the same way
+as you create virtual environments using python or conda.
+
+## What is the PythonSlicer executable?
+
+`PythonSlicer` is an executable provided in the `bin` directory of the Slicer installation.
+It’s the same version of Python available through the Slicer application, but it acts like
+a normal `python` (or `python3`) executable in terms of running scripts and can be used t
+ access all the packages installed in Slicer.
+
+This means you can use it in place of regular python to make use of the Slicer environment
+for batch processing and command-line operations.
+
+:::{warning}
+The application context is **not** available. This means that many objects like the running
+application, the MRML scene and the loaded modules are not available.
+:::
+
+:::{tip}
+You may use {func}`slicer.util.pip_install()` to install additional packages.
+:::
+
+## What is the Python Console?
+
+This describes the [](/user_guide/user_interface.md#python-console) available through
+the [user interface](/user_guide/user_interface.md).
+
+
+It allows to access to the running application context including the application instance,
+the MRML scene and all the loaded modules through these :mod:`slicer` attributes:
+* {attr}`slicer.app`
+* {attr}`slicer.mrmlScene`
+* {attr}`slicer.modules`
+* {attr}`slicer.moduleNames`
+
+:::{hint}
+Running scripts (or code) using the command-line option `--python-script` (or `pythoin-code`) is equivalent
+to running code in the Python console.
+:::
+
+:::{tip}
+You may use {func}`slicer.util.pip_install()` to install additional packages.
+:::
+
+:::{versionchanged} 5.2.0
+The `Python interactor` was renamed to `Python Console`.
+See [related discussion](https://discourse.slicer.org/t/rename-python-interactor-to-python-console/25897).
+:::
+
+## How to access Slicers Python API via an external program while Slicer is running?
+
+There are several ways to access Slicer's Python API from an external program while Slicer is running:
+
+- [SlicerWeb](/user_guide/modules/webserver.md): Exposes Slicer's Python environment as a web service
+  that can respond to http(s) requests with data from the current application state or modify the
+  application state. This is well-suited for applications that already use web requests.
+
+- [SlicerOpenIGTLink](https://github.com/openigtlink/SlicerOpenIGTLink): A lightweight socket-based
+  protocol for real-time data transfer. This is useful for applications that need to send many
+  requests per second (e.g., continuous data streaming) or for clients that only have access to
+  sockets and don't want to reimplement complex protocols like http. In most cases, it performs well
+  for sending requests in the range of 10 to 100 requests per second.
+  There is a native Python implementation: [pyigtl](https://pypi.org/project/pyigtl/).
```

Comment:
Can you describe if you would use `pyigtl` in addition to SlicerOpenIGTLink or as an alternative to SlicerOpenIGTLink? Any reason for picking one over the other?  Maybe provide info regarding if SlicerOpenIGTLink can be manipulated through python as well even though OpenIGTLinkIF is originally written in C++.

Reply:
In Slicer SlicerOpenIGTLink should be used (pyigtl just uses Python threading, which is not really parallelization).
pyigtl can be used everywhere else.

Reply:
Since this is Slicer documentation, does it make sense to include essentially general OpenIGTLink usage here? It seems like different uses of IGTLink whether with Slicer or some pure Python implementation might be better in the OpenIGTLink documentation.

Reply:
I agree that conceptually it is redundant to write about these interprocess communication options in the Slicer core documentation, but it has huge practical value to see a summary of all options at one place (similarly how we write about extensions that provide registration modules or DICOM plugins). The descriptions are fairly high-level, so they should not require a lot of maintenance.